parser-typechecker/src/Unison/Codebase/SqliteCodebase.hs

```diff
@@ -375,6 +397,13 @@ sqliteCodebase debugName root localOrRemote migrationStrategy action = do
     runTransaction action =
       withConn \conn -> Sqlite.runTransaction conn action
 
+    handleLockOption ma = case lockOption of
```

Comment:
I went with `handleLockOption` instead of `withCodebaseLock` or similar because a lock is not necessarily obtained, only if the lock option requests it. That said, I'm fine with naming it whatever.